src/gui/banlistoptionsdialog.cpp

```diff
@@ -107,7 +107,9 @@ void BanListOptionsDialog::on_buttonBanIP_clicked()
 
 void BanListOptionsDialog::on_buttonDeleteIP_clicked()
 {
-    const QModelIndexList selection = m_ui->bannedIPList->selectionModel()->selectedIndexes();
+    QModelIndexList selection = m_ui->bannedIPList->selectionModel()->selectedIndexes();
+    //sort indices in descending order to prevent shift of undeleted entries while deleting
+    std::sort(selection.rbegin(), selection.rend());
```

Comment:
Why you omit providing a custom compare function?
Also I really dislike using reverse iterator here.

Reply:
`QMmodelIndex` have builtin `operator<` but lack `operator>` for using `std::greater<>`.
And I dislikes lambda more then reverse iterators. And reverse iterators need less to code now  and to read later.

Reply:
>QMmodelIndex have builtin operator< but lack operator> for using std::greater<>.

Have you checked the documentation? There is nowhere mentioned that `QModelIndex::operator<` is guaranteed to use `row()` for comparison. You are making an unreasonable assumption here.

>And I dislikes lambda more then reverse iterators. 

Lambda is accepted in qbt codebase, no reason to avoid it when it is necessary.

>And reverse iterators need less to code now and to read later.

Right... and induce an potential performance hit due to how cache line work..

Reply:
> is guaranteed to use row()

I had checked source code. Comparison uses rows and columns and more internal members. I do not worry how exactly it works as far as it puts things in order. It is called *encapsulation* and *code reuse*.

>when it is necessary

But it is not necessary here, as you can see. AFAIR reverse iterators are not forbidden in qbt code guidelines too. Here it is just matter of taste.

> potential performance hit 

Premature optimization is the root of all evil. This code is definitely __not__ hot.

So this discussion can become something of holy-war kind. We need somebody else to help us to make decision. Preferably even number of votes should be used :)

Reply:
>It is called encapsulation and code reuse.

IMO you totally missed the point...

>Here it is just matter of taste.
>So this discussion can become something of holy-war kind. 

This is still far from an holy war debate. All opinion I presented none are subjective, all objective AFAIK.

>Premature optimization is the root of all evil. This code is definitely not hot.

Please don't get this wrong, this is not even called an optimization, it is merely "do the things right". Misinterpreting that quote always leads to lots of unnecessary performance wastes.
Also in C++ I believe there is no real "premature optimization", there are only lazy coders and bad design/decisions.

>We need somebody else to help us to make decision. Preferably even number of votes should be used :)

OK, since you are just waiving my review I'll just submit my version in another PR.

Reply:
>IMO you totally missed the point...

May be. I can not understand why it is so crucial to know the we are sorting exactly on rows()? What does that knowledge give us? Why we should not trust library authors on how do they do comparison of their indices? Would you use lambda comparator instead of library provided one if we have to sort in ascending order, so it can be done without reverse iterators? May be you have some solid objections against reverse iterators?

>"do the things right"

Starting thinking on cache lines behavior on code which would be run couple times per week does not looks like doing things right for me. I suppose to exclude performance considerations as soon as we don't use bubble sorting here.

Reply:
:\ You basically ignored all of my arguments and requested for more. There isn't any and sorry I won't be continuing this discussion.

Reply:
Well. speaking shortly:
1. I omit custom comparator because of this can be easily done. 
Pros: less code - less maintenance cost. And maintenance cost is most important reason in this case I think.
Contra: using reverse iterator. Somebody can overlook 'r' prefix. There is a comment for highlight reverse order as a countermeasure.

2. Performance on cache line level. Just take into account that this list is freshly returned from previous call to `selectedIndexes()`. That means that data is already in cache. Also mention that indices are stored in QList. Qlist internally keeps pointers to QModelIndex allocated on heap. So position of accessed data is not guarantied to be continuous. So it does not make much difference how would you access randomly placed already cached data - using forward, backward or random-like (as sort() actualy do) approach.